added new modals

FVuk opened the pull request:
- added modal-window (still doesn't work for some reason, plase run code review)
- added modal-team (only need to add our photos/avatars and LI pages)
- removed some unused features
- cleaned up refs.js, local-storage.js and index.js

FVuk committed:
added new modals

- added modal-window (still doesn't work for some reason, plase run code review)
- added modal-team (only need to add our photos/avatars and LI pages)
- removed some unused features
- cleaned up refs.js, local-storage.js and index.js

benbrzezinski commented on src/js/index.js:
```diff
@@ -1,5 +1,13 @@
+import './my-library.js';
+import './scrollTop.js';
+import './search.js';
+import './renderMovieCards.js';
```
Nie chcemy wszystkiego tak importować bo większość skryptów działa niezależnie od głównego pliku tylko są funkcjami pomocniczymi do większych funkcji, które np. trzeba tylko wywołać bo jak zobaczysz to znikła paginacja ze strony

FVuk replied:
naprawione, dodałem tylko modalTeam a rezta cofnąłem tak jak było więc paginacja znowu działa

benbrzezinski commented on src/js/local-storage.js:
```diff
@@ -3,33 +3,33 @@ const KEY_QUEUE = 'queue';
 export const watched = getWatchedLocalStorage() || [];
 export const queue = getQueueLocalStorage() || [];
```
Jak już zmieniłeś nazwy funkcji w localStorage to te dwie zostały bez zmian i wywalają undefined

FVuk replied:
poprawiłem wszystkie nazwy plików js na camelCase także zostawiam tylko kwestie modalWindow

benbrzezinski commented on src/js/api.js:
```diff
@@ -65,3 +65,4 @@ const Api = {
 };
 
 export default Api;
+export const API_KEY_IMG = `https://image.tmdb.org/t/p/original`;
```
Takie coś nie robimy można to dorzucić do modułu Api (lub zrobić to bez zmiennej dać ten url do template string) ale nie jako oddzielny export, ponieważ jak użyłeś dalej w kodzie takie coś `Api.API_KEY_IMG` to to nie zadziała jak zrobisz console.log tego to wywali undefined

FVuk replied:
dodałem do Api

benbrzezinski commented on src/js/modal-window.js:
```diff
@@ -0,0 +1,208 @@
+    cardForModal.innerHTML = `
+    <div class="modal__poster-thumb">
+          <img class="modal__poster" src="${
+            Api.API_KEY_IMG
```
To właśnie wywali undefined

benbrzezinski commented on src/sass/utils/_visually-hidden.scss:
```diff
@@ -1,15 +1,16 @@
 
-.is-hidden {
-  opacity: 0;
-  pointer-events: none;
```
Tu taka mała uwaga tą klasę wykorzystywała funkcja dla loader jak i svg właśnie loader, który też miał transition opacity ustawione, którego teraz nie ma. Jeśli z jakiegoś powodu potrzebujesz kolejną klasę, która w inny sposób coś znika niż po prostu opacity to dodaj kolejną pod inną nazwą jak np.
.visually-hidden {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 0;
  overflow: hidden;
  white-space: nowrap;
  border: 0;
  clip: rect(0 0 0 0);
  clip-path: inset(50%);
}

FVuk replied:
specjalnie sprawdzalem ale nie zauważyłem ze loader z tego korzysta. W kazdym razie podmieniłem klasy. is-hidden wykorzystuje modalTeam a .visually-hidden loader także wszystko działa

benbrzezinski commented on src/sass/utils/_modal-window.scss:
```diff
@@ -1,85 +1,308 @@
+  }
+}
+
+modal__info-thumb {
```
Brakuje kropki w klasie

FVuk replied:
fixed

benbrzezinski commented on src/sass/utils/_modal-window.scss:
```diff
@@ -1,85 +1,308 @@
+}
+
+.modal__poster-thumb {
+  display: flex;
+}
+.modal__poster {
```
Brak przerwy

FVuk replied:
fixed

benbrzezinski reviewed (changes requested):
Tak ogólnie to:
- nazwy plików js miały nazwy camelCase, czyli np. zamiast local-storage -> localStorage 
- modal z nami fajnie działa, a do kart z filmami, sugestia - osobiście użył bym biblioteki np. basiclightbox jak i tak mamy ją zainstalowaną przez npm, trzymałbym się dokumentacji i na podstawie tego napisał funkcję, te małe funkcję dałbym do innego modułu i importował do głównej funkcji

Edit: Jak z modalem karty filmu będzie dalej problem możesz wysłać zmiany, które dodają tylko ten team modal, a ja mogę zająć się films modal

FVuk committed:
added fixes in various files

- fixed camelCase in js files
- added API_KEY_IMG to Api
- added separate class for loader
- fixed modal-window.scss (missing dot and space)

benbrzezinski committed:
Merge branch 'main' into FVuk/feature/window-team-modals
